plugins/optimization-detective/storage/data.php

```diff
@@ -85,9 +85,49 @@ function od_get_normalized_query_vars(): array {
 	return $normalized_query_vars;
 }
 
+/**
+ * Get the URL for the current request.
+ *
+ * This is essentially the REQUEST_URI prefixed by the scheme and host for the home URL.
+ * This is needed in particular due to subdirectory installs.
+ *
+ * @since n.e.x.t
+ * @access private
+ *
+ * @return string Current URL.
+ */
+function od_get_current_url(): string {
+	$parsed_url = wp_parse_url( home_url() );
+	if ( ! is_array( $parsed_url ) ) {
+		$parsed_url = array();
+	}
+
+	if ( empty( $parsed_url['scheme'] ) ) {
+		$parsed_url['scheme'] = is_ssl() ? 'https' : 'http';
+	}
+	if ( ! isset( $parsed_url['host'] ) ) {
+		// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+		$parsed_url['host'] = isset( $_SERVER['HTTP_HOST'] ) ? wp_unslash( $_SERVER['HTTP_HOST'] ) : 'localhost';
+	}
+
+	$current_url = $parsed_url['scheme'] . '://' . $parsed_url['host'];
+	if ( isset( $parsed_url['port'] ) ) {
+		$current_url .= ':' . $parsed_url['port'];
+	}
+	$current_url .= '/';
+
+	if ( isset( $_SERVER['REQUEST_URI'] ) ) {
+		// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+		$current_url .= ltrim( wp_unslash( $_SERVER['REQUEST_URI'] ), '/' );
+	}
```

Comment:
Yes, `REQUEST_URI` should include the full path including the subdirectory root path. Otherwise, this function here only uses `home_url()` to obtain the default `scheme`, `host`, and `port`. The `path` is not used.